v2/offset_reader.go

```diff
@@ -0,0 +1,59 @@
+package car
+
+import "io"
+
+// OffsetReader implements Read, and ReadAt on a section
+// of an underlying io.ReaderAt.
+// The main difference between io.SectionReader and OffsetReader is that
+// NewOffsetReader accepts zero as n, the number of bytes to read, with
+// the trade-off that it does not implement Seek.
+// When n is set to zero, it will delegate io.EOF errors to the underlying
+// io.ReaderAt.
+// This is useful when reading a section at the end of a io.ReaderAt without
+// having to know the total number readable of bytes.
+type OffsetReader struct {
```

Comment:
I thought we just needed this to implement io.ReaderAt, not also io.Reader?

If we were to propose it for inclusion in the io package it would likely need both, but I think here we just need one method.

also, you could assert it, like:

    var _ io.ReaderAt = (*OffsetReader)(nil)

Reply:
Do you think it's worth removing the `io.Reader` implementation considering its small size and the fact that it is most likely needed when we propose this for inclusion in io package, despite its under-utilisation here?

Reply:
Up to you :)